ios/swift/AmpliSwiftSampleApp/Shared/ContentView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  ContentView.swift
+//  Shared
+//
+//  Created by Qingzhuo Zhen on 11/12/21.
+//
+
+import SwiftUI
+import CoreData
+
+struct ContentView: View {
+    @Environment(\.managedObjectContext) private var viewContext
+
+    @FetchRequest(
+        sortDescriptors: [NSSortDescriptor(keyPath: \Item.timestamp, ascending: true)],
+        animation: .default)
+    private var items: FetchedResults<Item>
+    private var ampli: Ampli = Ampli()
```

Comment:
I think we need a mechanism to provide a shared instance of Ampli. In JS/TS we export the default singleton instance as `ampli`, for swift I think we will need a dedicated getter e.g. `Ampli.getInstance()` or `Ampli.instance` to simplify the "common" case of a user wants to access a single instance across the entire application without having to manage the instance themselves. What do you think?

Reply:
Added `Ampli.instance` as the default value, updated in comment. `getInstance` is more like we can support multiple single instances like how SDK does. Not sure if we want to support that though.

I left this line as it is now since the example later used load method. And one minor thing is load method now can be called multiple times and different LoadOptions can cause amplitude client to be updated differently.. Seems same for node as well. Not sure if this is really ideal though.. Maybe we should add a check on load method first?

Reply:
I just added a check after disabled, so load method can overload disabled, but not amplitude instance. Let me know your thoughts. Also support get amplitude client from ampli.

Reply:
Cool, I think for now we can do a single `Ampli.instance` like you added. My hope/feeling is this is the 80+% case of users. If a user wants multiple instances they can still create them, however they need to manage the additional instances themselves. So 80% of users have a more concise interface, but power users can still get creative.

Not sure the user will want or need multiple instances of particular Ampli SDK. I think it's more likely that they would multiple tracking plans each with their own Ampli SDK wrapper instead. In this case `getInstance` wouldn't really help since there are two different Ampli classes.

Thanks for adding the initialize check in load, this is what we did in Iteratively where calling `load()` more than once would throw an error. I had intentionally omitted it here as some users had asked for the ability to call load multiple times with different options.  I agree that this is not best practice and should be avoided if possible.

I like your idea to allow resetting disabled. In the Itly SDK there was no good way to toggle-off tracking after load which made GDPR less than ideal. In this case the user can worst case access the Amplitude Client directly to disable tracking if needed.

Let's discuss more in our 1:1 today.